src/API/FeedState.php

```diff
@@ -261,7 +262,13 @@ private function add_feed_registration_state( $result ) {
 				throw new \Exception( esc_html__( 'Could not get merchant info.', 'pinterest-for-woocommerce' ) );
 			}
 
-			if ( 'ACTIVE' !== $merchant['data']->product_pin_feed_profile->feed_status ) {
+			$feed = Base::get_merchant_feed( $merchant_id, $feed_id );
+
+			if ( ! $feed ) {
+				throw new \Exception( esc_html__( 'Could not get feed info.', 'pinterest-for-woocommerce' ) );
+			}
```

Comment:
This check is unnecessary, as the `get_merchant_feed()` method can only return a valid profile or throw an exception.

```suggestion
```